spec/lib/email/receiver_spec.rb

```diff
@@ -879,7 +879,17 @@ def create_post_reply_key(value)
 
     it "cap the number of staged users created per email" do
       SiteSetting.maximum_staged_users_per_email = 1
-      expect { process(:cc) }.to change(Topic, :count)
+      expect { process(:cc) }.to change(Topic, :count).by(1)
+        .and change(User, :count).by(1)
+      expect(Topic.last.ordered_posts[-1].post_type).to eq(Post.types[:moderator_action])
+    end
+
+    it "cap the number of staged users existing per email" do
+      Fabricate(:user, email: "discourse@bar.com", staged: true) # from
+      Fabricate(:user, email: "someone@else.com", staged: true) # to
+
+      SiteSetting.maximum_staged_users_per_email = 1
+      expect { process(:cc) }.to change(Topic, :count).and not_change(User, :count)
```

Comment:
Do we want to do the same in this test as well?

```suggestion
      expect { process(:cc) }.to change(Topic, :count).by(1).and not_change(User, :count)
```